archaius-core/src/main/java/com/netflix/config/DynamicBooleanProperty.java

```diff
@@ -23,14 +23,41 @@
  *
  */
 public class DynamicBooleanProperty extends PropertyWrapper<Boolean> {
+
+    protected volatile boolean primitiveValue;
+
     public DynamicBooleanProperty(String propName, boolean defaultValue) {
         super(propName, Boolean.valueOf(defaultValue));
+
+        // Set the initial value of the cached primitive value.
+        this.primitiveValue = chooseValue();
+
+        // Add a callback to update the cached primitive value when the property is changed.
+        this.prop.addCallback(new Runnable() {
+            @Override
+            public void run() {
+                primitiveValue = chooseValue();
+            }
+        });
     }
+
     /**
      * Get the current value from the underlying DynamicProperty
+     *
+     * @return
+     */
+    private boolean chooseValue() {
+        Boolean propValue = this.prop == null ? null : this.prop.getBoolean();
```

Comment:
This method could be just the previous body of the get method: return prop.getBoolean(defaultValue).booleanValue().  The boxing and unboxing that will occur when the property value changes will be off the hot-path and a minor overhead for clearer code.